chainer/links/loss/hierarchical_softmax.py

```diff
@@ -23,12 +30,17 @@ def get_paths(self):
     def get_codes(self):
         return self.codes
 
+    def get_parent2child(self):
+        return self.parent2child
+
     def parse(self, tree):
         self.next_id = 0
         self.path = []
         self.code = []
         self.paths = {}
         self.codes = {}
+        self.parent2child = {}
```

Comment:
I think that it is easy to use `collections.defaultdict(list)` like the following line. 
https://github.com/pfnet/chainer/blob/master/chainer/links/loss/hierarchical_softmax.py#L88

Using `numpy.ndarray` is difficult in `_parse() function` since its behavior `append to left or right node`.